src/utils/compat.ts

```diff
@@ -1,6 +1,27 @@
 import { PublicKey } from '@solana/web3.js';
-import { Address } from '@solana/kit';
+import {
+  Address,
+  SignatureDictionary,
+  Transaction,
+  TransactionPartialSigner,
+  TransactionPartialSignerConfig,
+  TransactionSigner,
+} from '@solana/kit';
 
 export function toLegacyPublicKey(address: Address): PublicKey {
   return new PublicKey(address.toString());
 }
+
+export function noopSigner(address: Address): TransactionSigner {
```

Comment:
`compat.ts` is for things with compatibility between web3.js and solana/kit - this is a pure solana/kit impl so does not belong here.

It is also not needed in the SDK - it should live in the examples directory somewhere instead